luigi/contrib/slack_notifications.py

```diff
@@ -0,0 +1,181 @@
+Create an incomming webhook on slack: https://api.slack.com/incoming-webhooks
+
+Setup the slack integration with the following code.
+This can be done in the same module where the luigi tasks are defined.
+
+.. code-block:: python
+
+    from luigi.contrib.slack_notifications import configure_slack
+    configure_slack()
+
+Use a configuration like the following on luigi.cfg
+
+.. code-block:: ini
+
+    [slack]
+    webhook = https://hooks.slack.com/services/something
+    events = ["START", "SUCCESS", "FAILURE"]
+    template = Luigi task {task_id} event {event}: {args} {kwargs}
+
+
+Alternatives:
+
+Another plugin with similar purpouse is luigi-slack (https://github.com/bonzanini/luigi-slack). The major differences between this and luigi-slack is that:
+
+- Inspired by email notifications (notifications.py): simple configuration though luigi.cfg (no need for python configuration)
+- Minimal changes to python code (only call configure_slack()), no need create a new wrapper around luigi.run.
+- This only depends on requests module (does not depend on slackclient).
+- Work both for python 2 and 3
+- Configurable template, per event, that can expose specific task properties
+
+'''
+
+
+import logging
+import luigi
+import os
+import json
+
+logger = logging.getLogger(__name__)
+
+try:
+    import requests
+except ImportError:
+    logger.warning("Module %s requires the python package 'requests'." % __name__)
+
+
+class slack(luigi.Config):
+    webhook = luigi.parameter.Parameter(
+        default='',
+        description='Webhook URL to send the notifications')
+    channel = luigi.parameter.Parameter(
+        default='',
+        description='Channel to send slack notifications to')
+    username = luigi.parameter.Parameter(
+        default='',
+        description='Slack username originating the message')
+    icon_emoji = luigi.parameter.Parameter(
+        default='',
+        description='Slack bot icon emoji')
+    events = luigi.parameter.ListParameter(
+        default=['START', 'FAILURE', 'SUCCESS'],
+        description='List of events to be handled as slack notifications')
+    template = luigi.parameter.Parameter(
+        default='Luigi task {task_id} event {event}: {args} {kwargs}',
+        description='Template message for event handling slack notifications')
+    event_templates = luigi.parameter.DictParameter(
+        default={},
+        description='Custom template message per event')
+
+
+class SafeGetAttr(object):
+    """
+    Wraps an object returning a default_value for undefined attributes
+    """
+    def __init__(self, obj, default_value=None):
+        self._obj = obj
+        self._default_value = default_value
+
+    def __getattr__(self, attr):
+        return getattr(getattr(self, '_obj'), attr, getattr(self, '_default_value'))
+
+
+class SlackNotifyHandling():
+    slack_hook_url = None
+
+    def __init__(self, slack_config):
+        self.slack_hook_url = os.getenv('SLACK_HOOK_URL', slack_config.webhook)
```

Comment:
Yes, good points. I guess what would be better is to reference a file with secrets (similar to [.pg_password](https://www.postgresql.org/docs/9.4/static/libpq-pgpass.html) or [kubernetes secrets](https://www.postgresql.org/docs/9.4/static/libpq-pgpass.html)). But yes, I just removed reading from env variables as there is no benefit other than flexibility.